### PR TITLE
Add snippets to Logging's javadoc, LoggingSnippets class and tests

### PR DESCRIPTION
**Pull request by mziccard**

### CODE REVIEW

File: gcloud-java-examples/src/main/java/com/google/cloud/examples/logging/snippets/LoggingSnippets.java [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+  public Page<LogEntry> listLogEntries(String filter) {
+    // [START listLogEntries]
+    Page<LogEntry> entries = logging.listLogEntries(EntryListOption.filter(filter));
+    Iterator<LogEntry> entryIterator = entries.iterateAll();
+    while (entryIterator.hasNext()) {
+      LogEntry entry = entryIterator.next();
+      // do something with the entry
+    }
+    // [END listLogEntries]
+    return entries;
```
Comment (lesv) [inline gcloud-java-examples/src/main/java/com/google/cloud/examples/logging/snippets/LoggingSnippets.java]: Is this still useful at the end?  Can I create another Iterator and get everything?

Comment (mziccard) [inline gcloud-java-examples/src/main/java/com/google/cloud/examples/logging/snippets/LoggingSnippets.java]: Yes, you can get a new iterator calling `entries.iterateAll` (this is in fact what I do in the tests).

**Comment by lesv** [comment]
LGTM

**Pull request merged**